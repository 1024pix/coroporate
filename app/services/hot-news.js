import Service from '@ember/service';
import { inject as service } from '@ember/service';

export default Service.extend({

  prismic: service(),
  news: null,

  async load() {
    const news = await this.prismic.getHotNews();
    this.set('news', news);
  },
});
