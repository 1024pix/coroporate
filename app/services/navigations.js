import Service from '@ember/service';
import { inject as service } from '@ember/service';

export default Service.extend({

  prismic: service(),
  nav: null,

  async load() {
    const nav = await this.prismic.getMenus();
    this.set('nav', nav);
  }
});
