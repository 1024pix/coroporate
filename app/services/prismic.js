import Service from '@ember/service';
import { Promise } from "rsvp";

export default Service.extend({

  getApi() {
    return new Promise(function(resolve, reject) {
      Prismic.api('https://pix-site.cdn.prismic.io/api/v2', function(error, api) {
        if (error) {
          reject(error);
        }
        resolve(api);
      });
    });
  },

  async getFaqMenu() {
    const api = await this.getApi();
    const document = await api.query(Prismic.Predicates.at('document.type', 'faq_menu'), { 'fetchLinks': 'faq_category.title' });
    return document.results[0];
  },

  async getFaqCategoryByUid(faqCategoryUid) {
    const api = await this.getApi();
    const document = await api.query(Prismic.Predicates.at('my.faq_category.uid', faqCategoryUid));
    return document.results[0];
  },

  async findFaqItemsByFaqCategory(faqCategory) {
    const api = await this.getApi();
    const faqItemIds = faqCategory.rawJSON.faq_items.map(document => document.faq_item.id);
    const documents = await api.getByIDs(faqItemIds);
    return documents.results;
  },

  async findJobs() {
    const api = await this.getApi();
    const documents = await api.query(Prismic.Predicates.at('document.type', 'job'));
    return documents.results;
  },

  async getJobByUid(jobUid) {
    const api = await this.getApi();
    const document = await api.query(Prismic.Predicates.at('my.job_offer.uid', jobUid));
    return document.results[0];
  }

});
