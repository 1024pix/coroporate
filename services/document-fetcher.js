import LinkResolver from '~/plugins/link-resolver.js'

export const documents = {
  hotNews: 'hot_news',
  legalNotices: 'legal-notices',
  termsOfService: 'cgu_page',
  stats: 'statistiques',
  index: 'index',
  navigation: 'navigation',
  mainFooter: 'main_footer',
  mainNavigation: 'main_navigation',
  mediation: 'mediation',
  keyNumbers: 'key_numbers',
  employers: 'employers',
  schoolEducation: 'school-education',
  skills: 'competences',
}

export function documentFetcher(
  prismic,
  i18n = { defaultLocale: 'fr-fr' },
  req
) {
  const lang = i18n.locale || i18n.defaultLocale
  return {
    get: getSingle,
    findByType: async (type) => {
      const documents = await prismic.api.query(
        prismic.predicates.at('document.type', type),
        { lang }
      )
      return documents.results
    },
    getEmployers: async () => {
      const document = await prismic.api.getSingle('employers', {
        lang,
        fetchLinks: [
          'distributor_item.description',
          'distributor_item.footer',
          'distributor_item.link',
          'distributor_item.logo',
          'distributor_item.name',
        ],
      })
      return document
    },
    findNewsItems: async ({ page, pageSize } = { page: 1, pageSize: 20 }) => {
      const documents = await prismic.api.query(
        prismic.predicates.at('document.type', 'news_item'),
        {
          page,
          pageSize,
          orderings: '[my.news_item.date desc]',
          lang,
        }
      )
      return documents.results
    },
    getNewsItemByUid: async (slug) => {
      const document = await prismic.api.query(
        prismic.predicates.at('my.news_item.uid', slug),
        { lang }
      )
      return document.results[0]
    },
    getPreviewUrl: (previewToken) => {
      return prismic.api.previewSession(previewToken, LinkResolver, '/')
    },
    getPageByUid: async (uid) => {
      const simplePage = await getSimplePageByUid(uid)
      const slicePage = await getSlicesPageByUid(uid)
      return simplePage || slicePage
    },
  }

  function getSingle(type) {
    return prismic.api.getSingle(type, { lang })
  }

  async function getSimplePageByUid(uid) {
    const document = await prismic.api.query(
      prismic.predicates.at('my.simple_page.uid', uid),
      { lang }
    )
    return document.results[0]
  }

  async function getSlicesPageByUid(uid) {
    const document = await prismic.api.query(
      prismic.predicates.at('my.slices_page.uid', uid),
      { lang }
    )
    return document.results[0]
  }
}
