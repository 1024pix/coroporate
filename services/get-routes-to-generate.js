import prismic from 'prismic-javascript'
import _ from 'lodash'

export default async function () {
  const api = await prismic.getApi(process.env.PRISMIC_API_ENDPOINT)
  const { routes, totalPages } = await getRoutesInPage(api, 1)
  for (let page = 2; page <= totalPages; page++) {
    const { routes: nextPageRoutes } = await getRoutesInPage(api, page)
    routes.push(...nextPageRoutes)
  }
  return routes
}

async function getRoutesInPage(api, page) {
  const { results, total_pages: totalPages } = await api.query('', {
    pageSize: 100,
    page,
    lang: '*',
  })
  const uids = _.map(results, 'uid')
  const routes = _.reject(uids, _.isEmpty)
  return { totalPages, routes }
}
