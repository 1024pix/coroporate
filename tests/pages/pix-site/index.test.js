import { getInitialised } from './utils'
import { documentFetcher } from '~/services/document-fetcher'

jest.mock('~/services/document-fetcher')

describe('Index Page', () => {
  let wrapper

  beforeEach(async () => {
    documentFetcher.mockReturnValue({
      get: () =>
        Promise.resolve({
          data: {
            id: '',
            meta: '',
            body: [{}, {}, {}, {}, {}, {}, {}, {}],
          },
        }),
    })
    wrapper = await getInitialised('index')
  })
  test('mounts properly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders properly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
