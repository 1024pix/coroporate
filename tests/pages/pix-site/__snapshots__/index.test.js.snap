// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page get the corrects meta 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;

exports[`Index Page renders properly 1`] = `
"<div class=\\"index\\">
  <slice-zone-stub slices=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\"></slice-zone-stub>
</div>"
`;
